src/Symfony/Bridge/PhpUnit/Tests/DeprecationErrorHandler/DeprecationTest.php

```diff
@@ -49,9 +51,113 @@ public function testItRulesOutFilesOutsideVendorsAsIndirect()
         $this->assertFalse($deprecation->isIndirect());
     }
 
+    public function providerIsSelf(): array
+    {
+        return [
+            [true, '', 'MyClass1', ''],
+            [false, '', 'MyClass1', 'dummy_vendor_path'],
+            [
+                true,
+                serialize(['class' => '', 'method' => '', 'deprecation' => '', 'files_stack' => []]),
+                SymfonyTestsListenerForV5::class,
+                '',
+            ],
+            [
+                true,
+                serialize([
+                    'class' => '',
+                    'method' => '',
+                    'deprecation' => '',
+                    'triggering_file' => '',
+                    'files_stack' => [],
+                ]),
+                SymfonyTestsListenerForV5::class,
+                '',
+            ],
+            [
+                false,
+                serialize([
+                    'class' => '',
+                    'method' => '',
+                    'deprecation' => '',
+                    'triggering_file' => 'dummy_vendor_path',
+                    'files_stack' => [],
+                ]),
+                SymfonyTestsListenerForV5::class,
+                '',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider providerIsSelf
+     */
+    public function testIsSelf(bool $expectedIsSelf, string $message, string $traceClass, string $file): void
+    {
+        $trace = [
+            ['class' => 'MyClass1', 'function' => 'myMethod'],
+            ['class' => $traceClass, 'function' => 'myMethod'],
+        ];
+        $deprecation = new Deprecation($message, $trace, $file);
+        $this->assertEquals($expectedIsSelf, $deprecation->isSelf());
+    }
+
+    public function providerIsIndirectUsesRightTrace(): array
+    {
+        $reflexion = new \ReflectionClass(ClassLoader::class);
+        $vendorDir = dirname($reflexion->getFileName(), 2);
+
+        return [
+            [false, '', [['function' => 'myfunc1'], ['function' => 'myfunc2']]],
+            [
+                true,
+                '',
+                [
+                    ['function' => 'myfunc1', 'file' => $vendorDir.'/composer/ClassLoader.php'],
```

Comment:
Not that `composer` gets a special treatment in `isIndirect`, not sure if this is what you are testing for. If not, that's probably not the best file to pick if you want to avoid confusion.